modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/TypeScriptAngular2ClientCodegen.java

```diff
@@ -114,25 +115,33 @@ public String getTypeDeclaration(Property p) {
             MapProperty mp = (MapProperty)p;
             inner = mp.getAdditionalProperties();
             return "{ [key: string]: " + this.getTypeDeclaration(inner) + "; }";
+        } else if(p instanceof FileProperty || p instanceof ObjectProperty) {
+            return "any";
         } else {
-            return p instanceof FileProperty ? "any" : super.getTypeDeclaration(p);
+            return super.getTypeDeclaration(p);
         }
     }
 
     @Override
     public String getSwaggerType(Property p) {
         String swaggerType = super.getSwaggerType(p);
+        if(languageSpecificPrimitives.contains(swaggerType)) {
+            return swaggerType;
+        }
         return addModelPrefix(swaggerType);
     }
 
     private String addModelPrefix(String swaggerType) {
         String type = null;
         if (typeMapping.containsKey(swaggerType)) {
             type = typeMapping.get(swaggerType);
-            if (languageSpecificPrimitives.contains(type))
-                return type;
-        } else
+        } else {
+            type = swaggerType;
+        }
+
+        if (!languageSpecificPrimitives.contains(type)) {
```

Comment:
That would be a good workaround for now. Will add it when i have some time.
Personally I think it would be better to start using imports correctly in the future instead of prefixing everything with ```models```.